audio/out/ao_audiotrack.c

```diff
@@ -524,26 +527,37 @@ static int AudioTrack_write(struct ao *ao, int len)
 
 static void uninit_jni(struct ao *ao)
 {
-    JNIEnv *env = MP_JNI_GET_ENV(ao);
-    for (int i = 0; i < MP_ARRAY_SIZE(jclass_list); i++) {
-        mp_jni_reset_jfields(env, jclass_list[i].fields,
-                             jclass_list[i].mapping, 1, ao->log);
+    mp_mutex_lock(&jni_static_lock);
+    jni_static_use_count--;
+    if (jni_static_use_count == 0) {
+        JNIEnv *env = MP_JNI_GET_ENV(ao);
```

Comment:
a further safety check could be added here to ensure that two different mpv instances use the same `JNIEnv`.

Reply:
This should be fine. `Jclass` is a global object, and `JNIEnv` is a thread local object. As long as the `JNIEnv` use the same `JavaVM` behind, the `Jclass` object created through `NewGlobalRef` should be shared. And we can only have one `JavaVM` per android process.

Please check this document [JavaVM and JNIEnv](https://developer.android.com/training/articles/perf-jni#javavm-and-jnienv) 
> In theory you can have multiple JavaVMs per process, but Android only allows one.
> The JNIEnv is used for thread-local storage. For this reason, you cannot share a JNIEnv between threads.

Reply:
I'm aware it's a theoretical concern. We can leave this check out.